src/applications/mhv/secure-messaging/components/Search/FilterBox.jsx

```diff
@@ -0,0 +1,230 @@
+    const today = new Date();
+
+    if (range === DateRangeValues.LAST3) {
+      today.setMonth(today.getMonth() - 3);
+    } else if (range === DateRangeValues.LAST6) {
+      today.setMonth(today.getMonth() - 6);
+    } else if (range === DateRangeValues.LAST12) {
+      today.setMonth(today.getMonth() - 12);
+    }
+
+    return dateFormat(today, 'yyyy-MM-DD');
+  };
+
+  const checkFormValidity = () => {
+    // TODO: add validation for ALL blank fields
+    let formInvalid;
+    if (dateRange === 'custom' || testingDateRange) {
+      if (!fromDate && !testingFromDate) {
+        formInvalid = true;
+        setFromDateError('Please enter a start date');
+      }
+      if (!toDate && !testingToDate) {
+        formInvalid = true;
+        setToDateError('Please enter an end date');
+      }
+      if (
+        (fromDate || testingFromDate) &&
+        (toDate || testingToDate) &&
+        moment(toDate || testingToDate).isBefore(fromDate || testingFromDate)
+      ) {
+        formInvalid = true;
+        setFromDateError('Start date must be on or before end date');
+        setToDateError('End date must be on or after start date');
+      }
+    } else if (dateRange === 'any' && !category) {
+      formInvalid = true;
+      setFormError(true);
+    }
+    return formInvalid;
+  };
+
+  const handleFilterMessages = e => {
+    e.preventDefault();
+    const formInvalid = testingSubmit || checkFormValidity();
+    if (formInvalid) return;
+
+    const todayDateTime = moment(new Date()).format();
+    const offset = todayDateTime.substring(todayDateTime.length - 6);
+    let relativeToDate;
+    let relativeFromDate;
+    let fromDateTime;
+    let toDateTime;
+
+    if (
+      dateRange === DateRangeValues.LAST3 ||
+      dateRange === DateRangeValues.LAST6 ||
+      dateRange === DateRangeValues.LAST12
+    ) {
+      relativeToDate = moment(new Date());
+      relativeFromDate = `${getRelativeDate(dateRange)}T00:00:00${offset}`;
+    } else if (dateRange === DateRangeValues.CUSTOM) {
+      fromDateTime = `${fromDate}T00:00:00${offset}`;
+      toDateTime = `${toDate}T23:59:59${offset}`;
+    }
+
+    const folderData = folders.find(item => +item.id === +folderId);
+    dispatch(
+      runAdvancedSearch(folderData, {
+        category,
+        fromDate: relativeFromDate || fromDateTime,
+        toDate: relativeToDate || toDateTime,
+      }),
+    );
+    history.push('/search/results');
+  };
+
+  return (
+    <form
+      className="advanced-search-form filter-box"
+      onSubmit={handleFilterMessages}
+    >
+      {formError && (
+        <VaModal
+          modalTitle="Invalid search"
+          onPrimaryButtonClick={() => setFormError()}
+          onCloseEvent={() => setFormError()}
+          primaryButtonText="Ok"
+          status="error"
+          visible
+        >
+          <p>
+            Please use at least one of the following search fields or choose a
+            date range other than 'any'.
```

Comment:
Not sure if we want to continue displaying errors in modals as we a re switching to inline errors across the board, let's raise this with UCD during validation